src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/v1/features/FeatureService.java

```diff
@@ -407,26 +417,92 @@ public FeaturesResponse items(
             filters.add(FF.id(FF.featureId(itemId)));
         }
 
-        // The ids parameter is part of the draft proposal "Query by IDs". The syntax and semantic
-        // of the parameter is subject to change in a future release. Its usage should be carefully
-        // considered.
         if (ids != null && !ids.isEmpty()) {
-            filters.add(buildIdsFilter(ids));
+            // The ids parameter is part of the draft proposal "Query by IDs". The syntax and
+            // semantic
+            // of the parameter is subject to change in a future release. Its usage should be
+            // carefully
+            // considered.
+            if (featureServiceInfo.ids(wfsInfo)) {
+                filters.add(buildIdsFilter(ids));
+            } else {
+                LOGGER.warning(
+                        () ->
+                                "The ids parameter is not supported by the service, requires "
+                                        + FeatureConformance.IDS.getId()
+                                        + " conformance to be enabled.");
+            }
         }
-
         if (filter != null) {
-            Filter parsedFilter = filterParser.parse(filter, filterLanguage, filterCRS);
-            filters.add(parsedFilter);
+            CQL2Conformance cql2Info = CQL2Conformance.configuration(wfsInfo);
+            ECQLConformance ecqlInfo = ECQLConformance.configuration(wfsInfo);
+
+            if (featureServiceInfo.filter(wfsInfo)) {
+                if (filterLanguage == APIFilterParser.ECQL_TEXT && !ecqlInfo.text(wfsInfo)) {
+                    LOGGER.warning(
+                            () ->
+                                    "The filter language '"
+                                            + APIFilterParser.ECQL_TEXT
+                                            + "' is not supported by the service, requires "
+                                            + ECQLConformance.ECQL_TEXT.getId()
+                                            + " conformance to be enabled.");
```

Comment:
This bit is duplicated over and over and over... the pattern to report a disabled conformance class seems the same... suggest factoring out a common method. Should also make the code more readable, right now the logging is obfuscating what's going on in the method.

Reply:
resolved